src/libstd/sys/windows/args.rs

```diff
@@ -12,32 +12,98 @@
 
 use os::windows::prelude::*;
 use sys::c;
-use slice;
-use ops::Range;
 use ffi::OsString;
-use libc::{c_int, c_void};
 use fmt;
+use collections::VecDeque;
+use core::iter;
 
 pub unsafe fn init(_argc: isize, _argv: *const *const u8) { }
 
 pub unsafe fn cleanup() { }
 
 pub fn args() -> Args {
     unsafe {
-        let mut nArgs: c_int = 0;
-        let lpCmdLine = c::GetCommandLineW();
-        let szArgList = c::CommandLineToArgvW(lpCmdLine, &mut nArgs);
-
-        // szArcList can be NULL if CommandLinToArgvW failed,
-        // but in that case nArgs is 0 so we won't actually
-        // try to read a null pointer
-        Args { cur: szArgList, range: 0..(nArgs as isize) }
+        let lp_cmd_line = c::GetCommandLineW();
+        let mut parsed_args_list = parse_lp_cmd_line(lp_cmd_line as *const u16);
+        parsed_args_list.push_front(get_exe_name());
```

Comment:
Why is this unconditionally shoving the exe name in front? Isn't that already part of the command line? In your comparison test and the ReactOS code the exe name is set as the only argument when the command line was empty.

Reply:
You're right. That was a misunderstanding of what the test suite was telling me.